dashboard/test/controllers/api/v1/assessments_controller_test.rb

```diff
@@ -39,22 +50,188 @@ class Api::V1::AssessmentsControllerTest < ActionController::TestCase
+
+    # Student has completed an assessment.
+    level_source = create(
+      :level_source,
+      level: level1,
+      data: %Q({"#{sub_level1.id}":{"result":"This is a free response"},"#{sub_level2.id}":{"result":"0"},"#{sub_level3.id}":{"result":"1"},"#{sub_level4.id}":{"result":"-1"}})
+    )
+    create :activity, user: @student_1, level: level1,
+      level_source: level_source
+
+    updated_at = Time.now
+
+    create :user_level, user: @student_1, best_result: 100, script: script, level: level1, submitted: true, updated_at: updated_at, level_source: level_source
+
+    # Call the controller method.
+    get :section_responses, params: {
+      section_id: @section.id,
+      script_id: script.id
+    }
+
+    assert_response :success
+
+    # All these are translation missing because we don't actually generate i18n files in tests.
+    expected_response = {
+      @student_1.id.to_s => {
+        "student_name" => @student_1.name,
+        "responses_by_assessment" => {
+          level1.id.to_s => {
+            "stage" => "translation missing: en-US.data.script.name.#{script.name}.title",
+            "puzzle" => 1,
+            "question" => "Long assessment 1",
+            "url" => "http://test.host/s/#{script.name}/stage/1/puzzle/1?section_id=#{@section.id}&user_id=#{@student_1.id}",
+            "multi_correct" => 1,
+            "multi_count" => 4,
+            "submitted" => true,
+            "timestamp" => updated_at.utc.to_s,
+            "level_results" => [
+              {"student_result" => "This is a free response", "status" => "free_response"},
+              {"student_result" => "A", "status" => "correct"},
+              {"student_result" => "B", "status" => "incorrect"},
+              {"student_result" => "", "status" => "unsubmitted"},
+              {"status" => "unsubmitted"}
+            ]
+          }
+        }
+      }
+    }
+    assert_equal expected_response, JSON.parse(@response.body)
+  end
+
+  test "gets no anonymous survey data via assessment responses call" do
+    # Sign in as teacher and create a new script.
+    sign_in @teacher
+    script = create :script
+
+    # Set up an anonymous assessment in that script.
+    sub_level1 = create :text_match, name: 'level_free_response', type: 'TextMatch'
+    sub_level2 = create :multi, name: 'level_multi_unsubmitted', type: 'Multi'
+    sub_level3 = create :multi, name: 'level_multi_correct', type: 'Multi'
+    sub_level4 = create :multi, name: 'level_multi_incorrect', type: 'Multi'
+    create :multi, name: 'level_multi_unattempted', type: 'Multi'
+
+    level1 = create :level_group, name: 'LevelGroupLevel1', type: 'LevelGroup'
+    level1.properties['title'] =  'Long assessment 1'
+    level1.properties['anonymous'] = 'true'
+    level1.properties['pages'] = [{levels: ['level_free_response', 'level_multi_unsubmitted']}, {levels: ['level_multi_correct', 'level_multi_incorrect']}, {levels: ['level_multi_unattempted']}]
```

Comment:
nit: break this onto multiple lines

i think you could also set all of the `level1` properties in a single assignment like:
```
level1.properties = {
  title: 'Long assessment 1',
  anonymous: true,
  ...
}
```

Reply:
Chatted offline and decided this wasn't necessary right now.